statsmodels/stats/oaxaca.py

```diff
@@ -0,0 +1,254 @@
+        endog = np.column_stack((bi_col, endog))
+        bi = np.unique(bi_col)
+
+        # split the data along the bifurcate axis, the issue is you need to
+        # delete it after you fit the model for the total model.
+        exog_f = exog[np.where(exog[:, bifurcate] == bi[0])]
+        exog_s = exog[np.where(exog[:, bifurcate] == bi[1])]
+        endog_f = endog[np.where(endog[:, 0] == bi[0])]
+        endog_s = endog[np.where(endog[:, 0] == bi[1])]
+        exog_f = np.delete(exog_f, bifurcate, axis=1)
+        exog_s = np.delete(exog_s, bifurcate, axis=1)
+        endog_f = endog_f[:, 1]
+        endog_s = endog_s[:, 1]
+        endog = endog[:, 1]
+
+        self.gap = endog_f.mean() - endog_s.mean()
+
+        if swap and self.gap < 0:
+            endog_f, endog_s = endog_s, endog_f
+            exog_f, exog_s = exog_s, exog_f
+            self.gap = endog_f.mean() - endog_s.mean()
+
+        if hasconst is False:
+            exog_f = add_constant(exog_f, prepend=False)
+            exog_s = add_constant(exog_s, prepend=False)
+            exog = add_constant(exog, prepend=False)
+
+        self._t_model = OLS(endog, exog).fit(
+                                            cov_type=cov_type,
+                                            cov_kwds=cov_kwds)
+        self._f_model = OLS(endog_f, exog_f).fit(
+                                                cov_type=cov_type,
+                                                cov_kwds=cov_kwds)
+        self._s_model = OLS(endog_s, exog_s).fit(
+                                                cov_type=cov_type,
+                                                cov_kwds=cov_kwds)
+
+        self.exog_f_mean = np.mean(exog_f, axis=0)
+        self.exog_s_mean = np.mean(exog_s, axis=0)
+        self.t_params = np.delete(self._t_model.params, bifurcate)
+
+    def three_fold(self):
+        """
+        Calculates the three-fold Oaxaca Blinder Decompositions
+
+        Parameters
+        ----------
+
+        None
+
+        Returns
+        -------
+        char_eff : float
+            This is the effect due to the group differences in
+            predictors
+        coef_eff: float
+            This is the effect due to differences of the coefficients
+            of the two groups
+        int_eff: float
+            This is the effect due to differences in both effects
+            existing at the same time between the two groups.
+        gap: float
+            This is the gap in the mean differences of the two groups.
+        """
+
+        self.char_eff = (
+                        (self.exog_f_mean - self.exog_s_mean)
+                        @ self._s_model.params)
+        self.coef_eff = self.exog_s_mean @ (self._f_model.params
+                                            - self._s_model.params)
+        self.int_eff = ((self.exog_f_mean - self.exog_s_mean)
+                        @ (self._f_model.params - self._s_model.params))
+
+        if self.suppress is False:
```

Comment:
It all looks good, but we dont' use a pattern like this.  This block and its sibling below should be removed from here.  

What you could do would be the returna result class that has a __repr__ that prints these values, so that something like

```
res = mod.three_fold()
res
```

would print

```
Oaxaca-Blinder Three-way Effect

Characteristic Effect: XXX
Coefficent Effect: YYY
Interaction Effect: ZZZ
```

Reply:
Results class might be a pain since the two method have very different output. So could just remove the print block.